python/dlr/counter/publisher.py

```diff
@@ -34,25 +32,19 @@ def __init__(self):
     def send(self, data):
         try:
             self.record_queue.put(data)
-            self.event.set()
         except queue.Full as e:
             logging.exception("ccm msg publisher queue full")
         except Exception as e:
             logging.exception("unable to record messages in queue", exc_info=True)
 
     def _process_queue(self):
-        while self.event.wait() and not MsgPublisher._stop_processing:
+        while True and not MsgPublisher._stop_processing:
             self.client.send(self.record_queue.get(block=False))
```

Comment:
`self.record_queue.get(block=False)`  will throw `empty exception` if block is set false.  This needs to be wrapped with try-catch in case of empty queue.